src/finalizer/finalizer.cpp

```diff
@@ -38,6 +38,39 @@ extern "C" HRESULT Initialize(int argc, wchar_t* argv[])
     return hr;
 }
 
+extern "C" HRESULT StrTrimBackslash(LPWSTR* ppwz, LPCWSTR wzSource)
+{
+    HRESULT hr = S_OK;
+    LPWSTR sczResult = NULL;
+
+    int i = lstrlenW(wzSource);
+
+    if (0 < i)
+    {
+        for (i = i - 1; i > 0; --i)
+        {
+            if (L'\\' != wzSource[i])
+            {
+                break;
+            }
+        }
+
+        ++i;
+    }
+
+    hr = StrAllocString(&sczResult, wzSource, i);
```

Comment:
How does the `StrAllocString` function work?  It seems that `i` will be the length of the string including the backslash here, does `StrAllocString` copy `i - 1` characters to the new string because of the null terminator?

Reply:
`StrAllocString` relies on another function called `AllocStringHelper` which includes the null terminator.

As for 6.0.4xx, I would potentially like this in 6.0.3xx unless that's going to go out of support early. Anyone doing scripted setup work on Win7 would potentially be blocked. It's likely the Win7 usage is low, so 6.0.4xx would probably be fine